snap/channel.go

```diff
@@ -0,0 +1,171 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snap
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/arch"
+	"github.com/snapcore/snapd/strutil"
+)
+
+var channelRisks = []string{"stable", "candidate", "beta", "edge"}
+
+// Channel identifies and describes completely a store channel.
```

Comment:
(nitpick) To my ears `Channel icompletely dentifies and describes a store channel.` sounds a tad more natural, i.e. changing the position of "completely".
`

Reply:
@chipaca opinion on this? I'm tempted to merge and adjust in the/a follow up in case

Reply:
`Channel identifies and completely describes` -> does not completely identify
`Channel completely identifies and describes` -> completely does both
`Channel completely identifies, and describes` -> probably both, maybe just 1st
`Channel identifies and describes completely` -> probably just the second, maybe both
`Channel identifies and describes, completely` -> both again

take your pick